annotations/pdf.go

```diff
@@ -1,43 +1,41 @@
 package annotations
 
 import (
-	"fmt"
+	"bytes"
 	"os"
 
-	"github.com/jung-kurt/gofpdf"
 	"github.com/juruen/rmapi/archive"
-	"github.com/juruen/rmapi/encoding/rm"
+	"github.com/unidoc/unipdf/v3/creator"
+	pdf "github.com/unidoc/unipdf/v3/model"
 )
 
 const (
-	RmX      = rm.Width
-	RmY      = rm.Height
-	A4X      = 210
-	A4Y      = 297
-	ratioA4X = float32(A4X) / float32(RmX)
-	ratioA4Y = float32(A4Y) / float32(RmY)
+	ratioA4X = float32(0.443)
+	ratioA4Y = float32(0.443)
 )
 
 type PdfGenerator struct {
 	zipName        string
 	outputFilePath string
 	options        PdfGeneratorOptions
+	pdfReader      *pdf.PdfReader
+	template       bool
 }
 
 type PdfGeneratorOptions struct {
 	AddPageNumbers bool
 }
 
-func CreatePdfGenerator(zipName, outputFilePath string, options PdfGeneratorOptions) PdfGenerator {
-	return PdfGenerator{zipName: zipName, outputFilePath: outputFilePath, options: options}
+func CreatePdfGenerator(zipName, outputFilePath string, options PdfGeneratorOptions) *PdfGenerator {
+	return &PdfGenerator{zipName: zipName, outputFilePath: outputFilePath, options: options}
 }
 
-func (p PdfGenerator) Generate() error {
+func (p *PdfGenerator) Generate() error {
 	file, err := os.Open(p.zipName)
 	if err != nil {
 		return err
 	}
-	defer file.Close()
+	defer func() { _ = file.Close() }()
```

Comment:
Never seen this before, by curiosity, what is the difference with a regular `defer file.Close()`?

Reply:
That's a way to explicitly indicate that we are ignoring an error reported by the `Close` function.